src/DesignCompile/CompileExpression.cpp

```diff
@@ -773,7 +773,8 @@ expr *CompileHelper::reduceExpr(any *result, bool &invalidValue,
   }
   expr *res =
       eval.reduceExpr(result, invalidValue, m_exprEvalPlaceHolder, pexpr);
-  return res;
+  // If loop was detected, drop the partially constructed new value!
+  return m_unwind ? nullptr : res;
```

Comment:
@alaindargelas Is this acceptable change to break the dependency of the output on max loop counter?